.github/workflows/python.yml

```diff
@@ -23,74 +23,37 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  # TODO: would there be a way of integrating these into a single matrix job?
-  # Note that the commands are slightly different atm.
-  linux:
-    runs-on: ubuntu-latest
-    steps:
-      - name: 📂 Checkout code
-        uses: actions/checkout@v2
-      - name: 💰 Cache
-        uses: Swatinem/rust-cache@v1
-      - uses: messense/maturin-action@v1
-        with:
-          manylinux: auto
-          command: build
-          args: --release -o dist -m prql-python/Cargo.toml
-      - name: Upload wheels
-        uses: actions/upload-artifact@v2
-        with:
-          name: wheels
-          path: dist
-
-  windows:
-    runs-on: windows-latest
+  build_python:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, macos-latest, windows-latest ]
     steps:
-      - name: 📂 Checkout code
+      - name: Checkout
         uses: actions/checkout@v2
-      - name: 💰 Cache
-        uses: Swatinem/rust-cache@v1
-      - uses: messense/maturin-action@v1
+      - name: Build Wheels
+        uses: ./.github/actions/build
         with:
-          command: build
-          args: --release --no-sdist -o dist -m prql-python/Cargo.toml
-      - name: Upload wheels
-        uses: actions/upload-artifact@v2
+          OS: ${{ matrix.os }}
+      - name: Setup Nox
+        uses: ./.github/actions/setup_nox
         with:
-          name: wheels
-          path: dist
-
-  macos:
-    runs-on: macos-latest
-    steps:
-      - name: 📂 Checkout code
-        uses: actions/checkout@v2
-      - name: 💰 Cache
-        uses: Swatinem/rust-cache@v1
-      - uses: messense/maturin-action@v1
-        with:
-          command: build
-          args: --release --no-sdist -o dist --universal2 -m prql-python/Cargo.toml
-      - name: Upload wheels
-        uses: actions/upload-artifact@v2
-        with:
-          name: wheels
-          path: dist
-
+          cache_number: 0
+      - name: Run tests
+        run: nox -s tests -f prql-python/noxfile.py
   release:
     name: Release
     runs-on: ubuntu-latest
-    # TODO: Upload PYPI_API_TOKEN, confirm how we release all crates together.
-    # if: "startsWith(github.ref, 'refs/tags/')"
-    if: false
-    needs: [macos, windows, linux]
+    if: "startsWith(github.ref, 'refs/tags/')"
```

Comment:
This should presumably be `false` until we upload the token?

Reply:
Yep I'm currently altering this. The problem is that `GitHub.ref` contains whatever triggered the build - so if we don't trigger on tags, then this will never evaluate to true.